text/0089-flexible-inflation.md

```diff
@@ -0,0 +1,260 @@
+| **Authors**     | Kian Paimani                                                                                             |
+
+## Summary
+
+This RFC proposes a new `pallet_inflation` to be added to the Polkadot runtime, which improves
+inflation machinery of the Polkadot relay chain in a number of ways:
+
+1. More transparent and easier to understand inflation logic
+2. Easier parameterization through governance
+3. Decoupled from the staking logic, should inflation and staking happen in two disjoint consensus
+   systems, as proposed
+   [RFC32](https://polkadot-fellows.github.io/RFCs/approved/0032-minimal-relay.html).
+
+## Motivation
+
+The existing inflation logic in the relay chain is suffers from a number of drawbacks:
+
+* It is dated, as the number of parachain slots (and consequently auctions) will soon no longer be a
+  factor in determining the inflation rate.
+* Is hard to parameterize through on-chain governance, as the only way to tweak the inflation amount
+  is through changing a particular function directly in the source code ([example in Polkadot
+  runtime](https://github.com/polkadot-fellows/runtimes/blob/cc3beb8b2337a65e879e1d739c33e30230888267/relay/polkadot/src/lib.rs#L743-L769)).
+* Is deeply intertwined with the staking system, which is not an ideal design. For example, if one
+  wishes to know the inflation amount, an [`Event` from the staking
+  system](https://polkadot.subscan.io/event?page=1&time_dimension=date&module=staking&event_id=erapaid)
+  has to be interpreted, which is counter-intuitive.
+* Given all of this complexity, implementing an alteration which suggested a fixed percentage of the
+  inflation to go to the treasury was also [not possible in an ergonomic
+  way](https://github.com/paritytech/polkadot-sdk/pull/1660).
+
+This RFC, as iterated above, proposes a new `pallet_inflation` that addresses all of the named
+problems. However, **this RFC does not propose any changes to the actual inflation rate**, but
+rather provide a new technical substrate (pun intended), upon which token holders can decide on the
+future of the DOT token's inflation in a more clear and transparent way.
+
+We argue that one reason why the inflation rate of Polkadot has not significantly change in ~4 years
+has been the complicated process of updating it. We hope that with the tools provided in this RFC,
+stakeholders can experiment with the inflation rate in a more ergonomic way. Finally, this
+experimentation can be considered useful as a final step toward fixing the economics of DOT in JAM,
+as proposed in the JAM grayparper.
+
+Within the scope of this RFC, we suggest deploying the new inflation pallet in a backwards
+compatible way, such that the inflation model does not change in practice, and leave the actual
```

Comment:
Does this mean `pallet-inflation` would still take into account parachain slots for the inflation calculation? Since crowdloans are deprecated and core time in, I am wondering the relevancy of current logic (i.e. we might hit the saturation point of 60 slots pretty easily). May be we can push a sensible (and conservative) change in the first iteration itself?

Reply:
We are parameterizing the ideal staking rate already which makes sense 👍.

Reply:
It could take the auctions into account, but that is not correct anymore. 

What I propose is to also alter the existing inflation rate to a fixed idea rate (eg. 75%), no longer taking the auctions into account. 

My wording here is confusing, I will correct it.